[util]: Remove Insight references in paywall

camus-code opened the pull request:
Closes #747 

This commit removes all the direct references to the Insight API in `util/paywall.go`. However, this commit *does* leave in place much of the structure for utilizing a backup paywall URL, should we decide to move to a new one in the future.

camus-code commented:
Yeah you're totally right about that. The reason for doing so was that I wasn't sure if we had plans to implement a new backup block explorer in the future, so I didn't want to alter the structure too much and create additional work if/when that time comes.

I'll go ahead and refactor this code to make sense with a single block explorer unless you have any objections

lukebp commented:
@camus-code Refactor away.

camus-code commented:
Ok I think the recent changes should cover everything @lukebp

I've removed all the code regarding the backup block explorer and also renamed some of the 'primary' block explorer code so it makes sense in the context of using a single block explorer

lukebp commented on util/paywall.go:
```diff
@@ -376,65 +324,40 @@ func FetchTxWithBlockExplorers(address string, amount uint64, txnotbefore int64,
 
-	// Try the primary (dcrdata) first.
-	txID, amount, err := fetchTxWithPrimaryBE(primaryURL, address, amount,
+	//Fetch transaction from dcrdata
```
```suggestion
	// Fetch transaction from dcrdata
```

lukebp commented on util/paywall.go:
```diff
@@ -376,65 +324,40 @@ func FetchTxWithBlockExplorers(address string, amount uint64, txnotbefore int64,
 	if err != nil {
 		log.Printf("failed to fetch from dcrdata: %v", err)
 	} else {
 		return txID, amount, nil
```
This `else` conditional is redundant.

lukebp commented on util/paywall.go:
```diff
@@ -458,73 +381,35 @@ func convertBEPrimaryTransactionToTxDetails(address string, tx BEPrimaryTransact
-// request fails the insight block explorer is tried.
+// provided wallet address from the dcrdata block explorer
 func FetchTxsForAddress(address string) ([]TxDetails, error) {
 	// Get block explorer URLs
```
```suggestion
	// Get block explorer URL
```

No longer multiple URLs.

lukebp commented on util/paywall.go:
```diff
@@ -458,73 +381,35 @@ func convertBEPrimaryTransactionToTxDetails(address string, tx BEPrimaryTransact
 		return nil, fmt.Errorf("invalid address %v: %v", addr, err)
 	}
-	dcrdataURL, insightURL, err := blockExplorerURLForAddress(address,
+	dcrdataURL, err := blockExplorerURLForAddress(address,
```
This is now short enough for a single line.

camus-code commented:
Thanks for the review @lukebp. Everything should be good to go with this latest commit

lukebp commented on util/paywall.go:
```diff
@@ -93,8 +79,7 @@ type TxDetails struct {
 var (
-	// ErrCannotVerifyPayment is emitted when a transaction cannot be verified
-	// because it cannot reach either of the block explorer servers.
+	// ErrCannotVerifyPayment is emitted when a transaction cannot be verified by dcrdata
```
80 column limit

lukebp commented on util/paywall.go:
```diff
@@ -93,8 +79,7 @@ type TxDetails struct {
-	// ErrCannotVerifyPayment is emitted when a transaction cannot be verified
-	// because it cannot reach either of the block explorer servers.
+	// ErrCannotVerifyPayment is emitted when a transaction cannot be verified by dcrdata
 	ErrCannotVerifyPayment = errors.New("cannot verify payment at this time")
```
Why is this error here if its never actually being used?

lukebp commented on util/paywall.go:
```diff
@@ -376,65 +324,39 @@ func FetchTxWithBlockExplorers(address string, amount uint64, txnotbefore int64,
+	txID, amount, err := fetchTxWithBE(explorerURL, address, amount,
 		txnotbefore, minConfirmations)
 	if err != nil {
 		log.Printf("failed to fetch from dcrdata: %v", err)
```
Why are we logging the error instead of returning it?

lukebp commented on util/paywall.go:
```diff
@@ -458,73 +380,34 @@ func convertBEPrimaryTransactionToTxDetails(address string, tx BEPrimaryTransact
+	// Fetch using dcrdata block explorer
+	dcrdataTxs, err := fetchTxsWithBE(explorerURL)
 	if err != nil {
 		log.Printf("failed to fetch from dcrdata: %v", err)
```
Why are we logging the error instead of returning it?

camus-code commented:
Fixed the nits you addressed @lukebp. Moved the log statements to use `fmt.Errorf` instead. Thanks for the catch on `ErrCannotVerifyPayment` as well; looks like that was a remnant left over from the insight code.

lukebp commented:
@camus-code I'm not sure why Travis didn't run, but its going to fail.  Run the `goclean.sh` script in the politeia directory to see why.

camus-code commented:
@lukebp Hm yeah just noticing Travis didn't run as well. `goclean.sh` is coming back correctly now with the latest commit; didn't realize log was its own package.

Travis still isn't building though - is that something I could be doing wrong?
